### PR TITLE
Merge request for past and current commits.

### PR DESCRIPTION
**Pull request by Bounce23**
The first few at the top are just the rectifications of your comments on my last merge conflicts. 
The last are code additions, with what is now verified curve proofs for Doppio.

### CODE REVIEW

**Comment by CPerezz** [comment]
Build is failing. I'll check the logs so you can solve the errors.

Also, it will be nice to add better titles and descriptions of the PR and what does it include or remove.

File: sage_codes/Safe Curve code [hunk 1]
```diff
@@ -428,4 +434,45 @@ sage: res
 /// Hexidecimal representation
 (11 : 23E4A96ED9E13AF0600EA21DB1007D9FB6E47BDD233BBBD82EB8E634305E0B1 : 1)
 
-/// Prime order of curve, l
\ No newline at end of file
+<<<<<<< HEAD
```
Comment (CPerezz) [inline sage_codes/Safe Curve code]: This is a result of a non-solved Merge. Choose the version that you want and remove this line.

File: sage_codes/Safe Curve code [hunk 2]
```diff
@@ -428,4 +434,45 @@ sage: res
...
+sage: P
+7237005577332262213973186563042994240857116359379907606001950938285454250989
+sage: q
+7237005577332262213973186563042994240708213731944783966342240242656345919288
+sage: t = q+1-P
+sage: t
+-148902627435123639659710695629108331700
+
+/// the principle
+=======
```
Comment (CPerezz) [inline sage_codes/Safe Curve code]: This is a result of a non-solved Merge. Choose the version that you want and remove this line.

File: sage_codes/Safe Curve code [hunk 3]
```diff
@@ -428,4 +434,45 @@ sage: res
...
+/// Prime order of curve, l
+>>>>>>> a6b4da7fa363e916eeb9969ebd9693aff1adeb6a
```
Comment (CPerezz) [inline sage_codes/Safe Curve code]: This is a result of a non-solved Merge. Choose the version that you want and remove this line.

File: src/backend/u64/mod.rs [hunk 4]
```diff
@@ -1,4 +1,3 @@
-#[cfg(not(any(feature = "u64_backend")))]
```
Comment (CPerezz) [inline src/backend/u64/mod.rs]: Why this is being removed? This is the exact reason why this PR is not compiling. 
Here we force the compiler to use the `u64_feature` if none is selected. 

Please, this line has to be added again, there is absolutely no reason to remove this.

**Comment by CPerezz** [comment]
The build is failing since it was removed the following compiler instruction:
`#[cfg(not(any(feature = "u64_backend")))]` found at `src/backend/u64/mod.rs `.

Is there any reason why this was removed? Since this instruction forces the compiler to use the `u64_backend` if no feature has been selected.

File: src/backend/u64/mod.rs (hunk 4 above)
Comment (Bounce23) [reply src/backend/u64/mod.rs]: The issue stated above have been addressed and fixed.

File: sage_codes/Safe Curve code (hunk 3 above)
Comment (Bounce23) [reply sage_codes/Safe Curve code]: The issue stated above have been addressed and fixed.

Comment (Bounce23) [reply sage_codes/Safe Curve code]: The issue stated above has been addressed and fixed.

File: sage_codes/Safe Curve code (hunk 2 above)
Comment (Bounce23) [reply sage_codes/Safe Curve code]: The issue stated above has been addressed and fixed.

File: sage_codes/Safe Curve code (hunk 1 above)
Comment (Bounce23) [reply sage_codes/Safe Curve code]: The issue stated above has been addressed and fixed.

File: src/backend/u64/mod.rs (hunk 4 above)
Comment (CPerezz) [reply src/backend/u64/mod.rs]: Thanks!

**Pull request merged**